mutablesecurity/solutions/implementations/clamav/code.py

```diff
@@ -0,0 +1,752 @@
+        " periodically for virus database definition updates, downloads,"
+        " installs them, and notifies clamd to refresh it's in-memory virus"
+        " database cache."
+    )
+    TEST_TYPE = TestType.OPERATIONAL
+    FACT = ActiveService
+    FACT_ARGS = ("clamav-daemon",)
+
+
+class InternetAccess(BaseTest):
+    IDENTIFIER = "internet_access"
+    DESCRIPTION = "Checks if host has Internet access."
+    TEST_TYPE = TestType.REQUIREMENT
+    FACT = InternetConnection
+
+
+class TestScan(BaseTest):
+    @staticmethod
+    @deploy
+    def create_test() -> None:
+        files.directory(
+            sudo=True,
+            name="Creating the test directory in /tmp",
+            path="/tmp/clamav_test",
+            present=True,
+        )
+
+        server.shell(
+            sudo=True,
+            name="Creating the file on which we'll test ClamAV",
+            commands=[
+                "echo 'X5O!P%@AP[4\PZX54(P^)7CC)7}$EICAR-STANDARD-ANTIVIRUS-TEST-FILE!$H+H*'"
+                " > /tmp/clamav_test/test.txt"
+            ],
+        )
+
+        server.shell(
+            sudo=True,
+            name=(
+                "Running the ClamAV scan with the remove option on the given"
+                " directory created"
+            ),
+            commands=[
+                "clamscan --remove"
+                f" --log={ScanLogLocation.get()} /tmp/clamav_test"
+            ],
+            success_exit_codes=[1],
+        )
+
+        files.directory(
+            sudo=True,
+            name="Removing the test directory in /tmp",
+            path="/tmp/clamav_test",
+            present=False,
+        )
+
+    class TestScanFact(FactBase):
+        @staticmethod
+        def command() -> str:
+            current_date = datetime.today().strftime("%Y:%m:%d")
+
+            check_command = (
+                f"cat {ScanLogLocation.get()} | grep -c '{current_date}' ||"
+                " true"
+            )
+
+            return check_command
+
+        @staticmethod
+        def process(output: typing.List[str]) -> bool:
+            return int(output[0]) != 0
+
+    IDENTIFIER = "test_scan"
+    DESCRIPTION = "Checks if ClamAV is working properly."
+    TEST_TYPE = TestType.SECURITY
+    TRIGGER = create_test
+    FACT = TestScanFact
+
+
+class TextLogs(BaseLog):
+    class GeneratedLogsFact(FactBase):
+        @staticmethod
+        def command():
+            return (
+                "cat /var/log/clamav/freshclam.log "
+                f" /var/log/clamav/clamav.log {ScanLogLocation.get()}"
+            )
+
+        @staticmethod
+        def process(output: typing.List[str]) -> str:
+            return "\n".join(output)
+
+    IDENTIFIER = "text_logs"
+    DESCRIPTION = "The logs generated by ClamAV and FreshClam."
+    INFO_TYPE = IntegerDataType
+    FACT = GeneratedLogsFact
+
+
+class StartScan(BaseAction):
+    @staticmethod
+    @deploy
+    def scan_device(scan_location: str) -> None:
+        server.shell(
```

Comment:
Don't forget adding the expected exit code. If a virus is detected, then the code will non-zero and MutableSecurity will think that there is an execution issue.

Reply:
Yep, almost forgot. Double checked every other scan also, and they should be fine now.